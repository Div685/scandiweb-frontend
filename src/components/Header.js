import React from 'react'

const Header = ({ handleSubmit, handleClear, title, firstBtn, secondBtn }) => {

  return (
    <header>
      <h3 className="title"> {title} </h3>
      <div className="header__button">
        <button type="submit" onClick={handleSubmit}> {firstBtn} </button>
        <button type="submit" onClick={handleClear} className="delete-checkbox"> {secondBtn} </button>
      </div>
    </header>
  )
}

export default Header
