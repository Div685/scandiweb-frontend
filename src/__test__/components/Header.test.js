import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Header from '../../components/Header';

const title = 'Product List';
const handleSubmit = jest.fn();
const handleClear = jest.fn();
const firstBtn = 'ADD';
const secondBtn = 'Delete';
beforeEach(() => {
  //   // setup a DOM element as a render target
  render(
    <Header 
      title={title}
      firstBtn={firstBtn}
      secondBtn={secondBtn}
      handleClear={handleClear}
      handleSubmit={handleSubmit}
    />
    );
});

describe('Header Component', () => {
  it('render Header Page correctly', () => {
    const tree = renderer.create(
      <Header 
      title={title}
      firstBtn={firstBtn}
      secondBtn={secondBtn}
      handleClear={handleClear}
      handleSubmit={handleSubmit}
    />,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Header ', () => {
  it('render Header Title', () => {
    expect(screen.getByRole('heading').textContent).toBe(' Product List ');
  });
});
