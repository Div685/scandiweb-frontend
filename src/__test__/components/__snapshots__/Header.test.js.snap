// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render Header Page correctly 1`] = `
<header>
  <h3
    className="title"
  >
     
    Product List
     
  </h3>
  <div
    className="header__button"
  >
    <button
      onClick={[MockFunction]}
      type="submit"
    >
       
      ADD
       
    </button>
    <button
      onClick={[MockFunction]}
      type="submit"
    >
       
      Delete
       
    </button>
  </div>
</header>
`;
